chore (refs T33323): add storybook documentation to DpTabs. (#344)
demos-europe/demosplan-ui: src/components/DpTabs/DpTabs.stories.mdx

import { Meta } from '@storybook/addon-docs'
import DpTabs from './DpTabs'

<Meta
  title="Components/Tabs"
  component={DpTabs} />

# DpTabs

> DpTabs is a wrapper for DpTab component through `<slot />`.
> 
> DpTabs component is a simple user interface (tabs UI) for organizing related content and occupying a compact space.
> 
> The DpTab components are aligned horizontally, and each tab is associated with its header.
> 
> Components DpTabs and DpTab contain `<slot />`.

Reviewer: ```suggestion
The DpTabs component provides a structured and interactive way to organize and navigate related content, allowing users to switch between different views while keeping context.

DpTabs and DpTab both provide a default slot. For each tab, a `DpTab` instance is placed inside the default slot of `DpTabs`, while the content of each tab is placed inside the default slot of the respective `DpTab` instance.

The `DpTab` instances are then aligned horizontally, and each tab is associated with its header.
```

## Default usage

```html
<dp-tabs
    tab-size="medium"
    :active-id="activeTabId"
    use-url-fragment

>     @change="setActiveTabId"> // call a method

Reviewer: Please use html comments here.

Author: Done in https://github.com/demos-europe/demosplan-ui/pull/344/commits/579d5acd26d7468522a98a237c8a945dc6411efd

    <dp-tab

>       id="id"

Reviewer: ```suggestion
      id="idTab1"
```

>       :label="Translator.trans('tab.name')">

Reviewer: To simplify the example i'd suggest to drop the translator here (we will eventually drop it off demosplan-ui completely):

```suggestion
      label="Tab Title 1">
```

      <slot>

>         <component-name />

Reviewer: To simplify the example i'd suggest to put plain text content here, instead of a component.

```suggestion
        Tab content 1.
```

      </slot>
    </dp-tab>
    <dp-tab

>       id="id"

Reviewer: ```suggestion
      id="idTab2"
```

>       :label="Translator.trans('tab.name')">

Reviewer: ```suggestion
      label="Tab Title 2">
```

      <slot>

>         <component-name />

Reviewer: ```suggestion
        Tab content 2.
```

      </slot>
    </dp-tab>
  </dp-tabs>
```

## DpTabs Props:

```
- activeId: is a String type, belongs to data.

- tabSize: `medium` or `large`, default is `large`

- useUrlFragment: Active tab state may be persisted via an Url fragment.
                  Also, ab clicks are pushed to the browser history to enable state change when browser navigation is used.
```

## DpTab Props:

```
- id: Tab ID.

- label: The label is passed to the DpTab child component, but then it is used in the parent component DpTabs.

- suffix: A html string may be passed here which is appended to the tab label.

> ```

Reviewer: Vue component props are typically "auto-documented" by storybook using [vue-docgen-api](https://vue-styleguidist.github.io/docs/Docgen.html) and vue-docgen-loader. At the moment the Storybook docs integraton is broken so that the props are not showing up as expected, however there is no need to document them by hand. Rather we could fill in the blanks by adding docblocks directly above the props themselves.

Author: Done https://github.com/demos-europe/demosplan-ui/pull/344/commits/59b2c5bed951c294cf74dc3000a75077ceaeb6ba
